### PR TITLE
Test: Implement unit tests for hook functionality and edge cases.

### PR DESCRIPTION
**Pull request by abdelfatah-ashour**

### CODE REVIEW

File: src/useNumberState.ts [hunk 1]
```diff
@@ -5,9 +5,9 @@ import { isNumeric } from "./utils";
 export function useNumberState(searchParamName: string, options: { defaultValue: number }) {
   const [searchParams, setSearchParams] = useSearchParams();
 
-  const acquiredSearchParam = searchParams.get(searchParamName);
+  const acquiredSearchParam = searchParams.get(searchParamName); // string | null
 
-  const finalValue = isNumeric(acquiredSearchParam) ? (Number(acquiredSearchParam) as number) : options.defaultValue;
+  const finalValue = acquiredSearchParam ? Number(acquiredSearchParam) : options.defaultValue;
```
Comment (abdelgawad1) [inline src/useNumberState.ts]: @abdelfatah-ashour This is not a bug, the original line checks if the value can be converted into a number then it parses it. 

case: the query string is `x=hello`, while we are using `useNumberState`. it will use `Number(hello)` as the value while it's not valid

File: src/utils.ts [hunk 2]
```diff
@@ -21,15 +21,8 @@ export function arraysAreIdentical<T>(arr1: T[], arr2: T[]): boolean {
   }
 
   // Sort both arrays
-  const sortedArr1 = [...arr1].sort();
-  const sortedArr2 = [...arr2].sort();
+  const sortedArr1 = [...arr1].sort().toString();
+  const sortedArr2 = [...arr2].sort().toString();
```
Comment (abdelgawad1) [inline src/utils.ts]: @abdelfatah-ashour with the new comparison `[null, undefined]` will be equal to `[undefined, null]` since both will be converted to `,`.
